### DISCUSSION TITLE
Using lnx for dynamic data search/ranking

### DISCUSSION THREAD

**Discussion by jvdwrf** [lnx-search/lnx#143] [Q&A] [answered]
Hi there!

I would like to use lnx for my next project since it looks very promising, however I'm not quite sure how well it would suit my goals.

Basically, I want to rank user-made posts based on (a combination of):
1. Tags of the post, which can be searched for.
2. Recency of posting.
3. Upvotes and downvotes.
4. Keyword/fuzzy search.

The tags and content of posts should be quite static elements, however the upvotes and downvotes of the posts will be very dynamic. I'm afraid that at the moment, I would have to delete posts and re-index them every time (or every x times) a post is up/downvoted. 

Would lnx be a good solution for such a platform, or would you recommend something else instead?

Thanks in advance!

**Comment by ChillFish8** [comment, marked as answer]
Hello 👋 

I think the biggest issue you would have is if you want to take *all* of those parameters into account for a single query, the current query API doesn't let you re-rank things based on the contents of a numeric field for example.

If you are trying to do those separately though, then it should be fine.

> he tags and content of posts should be quite static elements, however the upvotes and downvotes of the posts will be very dynamic. I'm afraid that at the moment, I would have to delete posts and re-index them every time (or every x times) a post is up/downvoted.

I would not worry too much about this unless you are looking at hundreds of thousands of updates per second, lnx will generally quite happily handle at least a few thousand updates a second without issue on limited resources.

**Reply by jvdwrf** [reply]
Okay perfect, thanks for the detailed answer.

I will try lnx out I think and see how it works for this use case hopefully.
